inconspiquous/dialects/qu.py

```diff
@@ -25,6 +31,16 @@ class BitType(ParametrizedAttribute, TypeAttribute):
     name = "qu.bit"
 
 
+@irdl_attr_definition
+class QReg(ParametrizedAttribute, TypeAttribute):
+    """
+    Type for a register of qubits of a static size.
+    """
+
+    name = "qu.reg"
+    size: ParameterDef[IntegerAttr]
```

Comment:
```suggestion
    size: ParameterDef[IntAttr]
```

I realise this is not what was written on the issue, but I think in retrospect it would be better for this to have an `IntAttr` parameter instead, which doesn't require an integer type, this would simplify some of the printing in the test you have added.